CODE_OF_CONDUCT.md

```diff
@@ -0,0 +1,83 @@
+## Goal
+
+Our goal is to provide a space where it is safe for everyone to contribute to
+open-source software in a respectful and peaceful manner.
+
+We value all contributions and want to make this organization and its surrounding community
+a place for everyone.
+
+As members, contributors and everyone else who may participate in the development, we strive
+to keep the entire experience civil.
+
+## Standards
+
+Our community is made for everyone to contribute to the codebase together.
+
+Our standards are:
+ - Do not harass, attack, or in any other way discriminate against anyone, including for their protected traits, including, but not limited to, sex, religion, race, appearance, gender, identity, nationality, sexuality, etc.
```

Comment:
Well, on the other hand, you could leave out a lot of stuff. For instance, this has no mention of disability, economic status, etc, but should anyone be harassed on those grounds? Closing this hole is why I suggested the change of "*including* for their protected traits", not to create new problematic protected traits, but to enumerate some traits which are protected and clarify that the harassment is harassment regardless of whatever traits it's based on.

Reply:
well that's how it's worded, no?
  
  Do not harass, attack, or in any other way discriminate against anyone, **including** for their protected traits,

Reply:
Yeah, yeah, this is good, I just think that the specific wording of "protected traits" being established here and elsewhere is a bit weasel worded. I'm just picking nits and suggesting minor editorial corrections, this is not a big deal.

Reply:
Maybe just leave out the "protected traits" part, leaving it clear and conscise.

> "Do not harass, attack, or in any other way discriminate against anyone."

Seems to me that this can not in any way be misunderstood.

Reply:
It's important to call out at least a few specific traits. Many communities (including hyprland prior to these changes, notably) don't consider harassment on the basis of these traits as a genuine form of harassment. It's important to signal that these groups are welcome here, specifically because they are unwelcome in many places by default.

Reply:
> and so on.

I would expand that list.

We could grab a few from the [GNOME code of conduct](https://wiki.gnome.org/Foundation/CodeOfConduct):

> age
> body size
> caste
> citizenship
> disability
> education
> ethnicity
> familial status
> gender expression
> gender identity
> genetic information
> immigration status
> level of experience
> nationality
> personal appearance
> pregnancy
> race
> religion
> sex characteristics
> sexual orientation
> sexual identity
> socio-economic status
> tribe
> veteran status

Reply:
but why? There is a strong "no harassment" point. You can make this argument forever, adding traits until the end of time. We put some traits as examples, not as an exhaustive list (note the _including_ and _etc._)

Reply:
Adding more specifics makes it more apparent to people that Hyprland isn't a place to propagate hate.

Reply:
It does, but at some point it clutters the CoC, and makes people less inclined to read it because of the length.

I believe we've covered enough common sources of specific targeted hate, and the rest fall under "etc".

"no harassment" is still "no harassment", IMO.

Reply:
Make it a requirement to sign a box in a new PR that says, “I read and agreed with the CoC”

Reply:
I don't think it's necessary to pick nits/bikeshed over the specific list of protected groups enumerated in this document.

Reply:
Fair enough. I find it difficult that it may give the impression that some groups are being excluded, even though that is not the intention